infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -462,10 +460,29 @@ protected void initChannel(@Nonnull Channel ch) throws Exception {
                     SSLEngine engine = sslContext.newEngine(ch.alloc());
                     engine.setEnabledCipherSuites(enabledTlsCipherSuites);
                     engine.setEnabledProtocols(enabledTlsProtocols);
+
                     if (tlsMutualAuthEnabled) {
                         engine.setNeedClientAuth(true);
                     }
-                    ch.pipeline().addLast("ssl", new SslHandler(engine));
+
+                    SslHandler sslHandler = new SslHandler(engine);
+
+                    // Add listener to handle SSL Handshake Exceptions in order to properly log
+                    // the target IP failing to authenticate against this server.
+                    sslHandler.handshakeFuture().addListener(channelFuture -> {
+                        if (channelFuture.cause() != null && channelFuture.cause().getClass()
+                                .equals(SSLHandshakeException.class)) {
+                            String peerHost;
+                            try {
+                                peerHost = ((java.net.InetSocketAddress) ch.remoteAddress()).getAddress().getHostAddress();
+                            } catch (Exception e) {
+                                peerHost = "Not determined.";
+                            }
+                            log.error("CorfuServer authentication failed for target IP: {}", peerHost);
```

Comment:
Can you revise the wording? The clients try to authenticate with the server so is the ip really a target?
Consider:
`Client authentication failed (remote address {address})`